### DISCUSSION TITLE
Allow filling of properties on an existing instance

### DISCUSSION THREAD

**Discussion by Maxwell175** [dadhi/DryIoc#603] [Q&A] [answered]
Hello,

I would like to inject services only for properties that have a specific attribute (I am using the MEF extensions for this). I would like to be able to inject the properties of an existing instance of a class that was created using some other mechanism than .Resolve().

```c#
class A
{
    [Import]
    public MyService Svc { get; set; }
}

var A = new A();
container.<some function>(A);
Assert.NotNull(A.Svc);
```

**Comment by dadhi** [comment, marked as answer]
@Maxwell175 You may use `InjectPropertiesAndFields` method:

```cs
container.InjectPropertiesAndFields(a)
```
